pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -312,67 +313,77 @@ func validateNetworksMatchInterfaces(field *k8sfield.Path, spec *v1.VirtualMachi
+	switch {
+	case !networkExists:
 		causes = appendStatusCauseForNetworkNotFound(field, causes, idx, iface)
-	} else if iface.Slirp != nil && networkData.Pod == nil {
+	case iface.Slirp != nil && networkData.Pod == nil:
 		causes = appendStatusCauseForSlirpWithoutPodNetwork(field, causes, idx)
-	} else if iface.Slirp != nil && networkData.Pod != nil && !config.IsSlirpInterfaceEnabled() {
+	case iface.Slirp != nil && networkData.Pod != nil && !config.IsSlirpInterfaceEnabled():
 		causes = appendStatusCauseForSlirpNotEnabled(field, causes, idx)
-	} else if iface.Masquerade != nil && networkData.Pod == nil {
+	case iface.Masquerade != nil && networkData.Pod == nil:
 		causes = appendStatusCauseForMasqueradeWithoutPodNetwork(field, causes, idx)
-	} else if iface.Masquerade != nil && link.IsReserved(iface.MacAddress) {
+	case iface.Masquerade != nil && link.IsReserved(iface.MacAddress):
 		causes = appendStatusCauseForInvalidMasqueradeMacAddress(field, causes, idx)
-	} else if iface.InterfaceBindingMethod.Bridge != nil && networkData.NetworkSource.Pod != nil && !config.IsBridgeInterfaceOnPodNetworkEnabled() {
+	case iface.InterfaceBindingMethod.Bridge != nil && networkData.NetworkSource.Pod != nil && !config.IsBridgeInterfaceOnPodNetworkEnabled():
 		causes = appendStatusCauseForBridgeNotEnabled(field, causes, idx)
-	} else if iface.InterfaceBindingMethod.Macvtap != nil && !config.MacvtapEnabled() {
+	case iface.InterfaceBindingMethod.Macvtap != nil && !config.MacvtapEnabled():
 		causes = appendStatusCauseForMacvtapFeatureGateNotEnabled(field, causes, idx)
-	} else if iface.InterfaceBindingMethod.Macvtap != nil && networkData.NetworkSource.Multus == nil {
+	case iface.InterfaceBindingMethod.Macvtap != nil && networkData.NetworkSource.Multus == nil:
 		causes = appendStatusCauseForMacvtapOnlyAllowedWithMultus(field, causes, idx)
-	} else if iface.InterfaceBindingMethod.Passt != nil && !config.PasstEnabled() {
+	case iface.InterfaceBindingMethod.Passt != nil && !config.PasstEnabled():
 		causes = appendStatusCauseForPasstFeatureGateNotEnabled(field, causes, idx)
-	} else if iface.Passt != nil && networkData.Pod == nil {
+	case iface.Passt != nil && networkData.Pod == nil:
 		causes = appendStatusCauseForPasstWithoutPodNetwork(field, causes, idx)
-	} else if iface.Passt != nil && numOfInterfaces > 1 {
+	case iface.Passt != nil && numOfInterfaces > 1:
 		causes = appendStatusCauseForPasstWithMultipleInterfaces(field, causes, idx)
-	} else if iface.Binding != nil && !config.NetworkBindingPlugingsEnabled() {
+	case iface.Binding != nil && !config.NetworkBindingPlugingsEnabled():
 		causes = appendStatusCauseForBindingPluginsFeatureGateNotEnabled(field, causes, idx)
 	}
 	return causes
 }
 
-func validateDHCPExtraOptions(field *k8sfield.Path, iface v1.Interface) (causes []metav1.StatusCause, done bool) {
+func validateDHCPExtraOptions(field *k8sfield.Path, iface v1.Interface) ([]metav1.StatusCause, bool) {
+	var causes []metav1.StatusCause
+	var done bool
```

Comment:
```suggestion
	done := false
```

To be consistent with `validateNetworksMatchInterfaces`?

Reply:
And to avoid L353